cloudinit/tests/test_util.py

```diff
@@ -387,6 +387,10 @@ def test_subp_exception_raises_to_caller(self, m_subp):
 @mock.patch('os.path.exists')
 class TestGetLinuxDistro(CiTestCase):
 
+    def setUp(self):
+        util.get_linux_distro.cache_clear()
```

Comment:
As you've noted in IRC, this will fail on Python 2 because our shim doesn't have `cache_clear` (thank you for helping me understand!). We could either add a noop `cache_clear` to the shim path, or we could make this something like:

```suggestion
        if hasattr(util.get_linux_distro, "cache_clear"):
            util.get_linux_distro.cache_clear()
```

and then the tests will work whether or not the function is decorated with `lru_cache()` or not. What do you think?